lib/stages/RawReader.hpp

```diff
@@ -22,9 +22,16 @@
 #include <utility>  // for pair
 #include <vector>   // for vector
 
+using nlohmann::json;
+
 /**
- * @class visRawReader
- * @brief Read and stream a raw visibility file.
+ * @class RawReader
+ * @brief Generic class to read and stream a raw file.
+ *
+ * All classes which inherit from this should provide the following API:
```

Comment:
This is another one where the hot potato landed on you, but this class could really do with a clear description of the read order. The description in the paragraph below isn't very clear at all. An example might make it clearer, also just saying *why* you might ever want to read in this chunked order.

Reply:
I'm not entirely sure of the motivation behind the chunking to write about it. I know it's to improve the reading efficiency. Could maybe you or @tristpinsm provide a brief summary I can add in?

Reply:
The reading is done in chunks for writing efficiency. The `visTranspose` stage writes to an HDF5 file with a chunked layout, and this means that IO is performed on entire chunks, so it is most efficient to write out on exact chunk boundaries. The reason for using chunking in the first place is to enable compression and try to optimise for certain IO patterns (i.e. read a few frequencies for many times).

Reply:
Thanks @tristpinsm!